test/powershell/Modules/Microsoft.PowerShell.Management/ControlService.Tests.ps1

```diff
@@ -0,0 +1,85 @@
+Describe "Control Service cmdlet tests" -Tags "Feature","RequireAdminOnWindows" {
+  BeforeAll {
+    $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+    if ( -not $IsWindows ) {
+        $PSDefaultParameterValues["it:skip"] = $true
+    }
+  }
+  AfterAll {
+    $global:PSDefaultParameterValues = $originalDefaultParameterValues
+  }
+
+  It "StopServiceCommand can be used as API for '<parameter>' with '<value>'" -TestCases @(
+    @{parameter="Force";value=$true},
+    @{parameter="Force";value=$false},
+    @{parameter="NoWait";value=$true},
+    @{parameter="NoWait";value=$false}
+  ) {
+    param($parameter, $value)
+    $stopservicecmd = [Microsoft.PowerShell.Commands.StopServiceCommand]::new()
+    $stopservicecmd.$parameter = $value
+    $stopservicecmd.$parameter | Should Be $value
+  }
+
+  It "RestartServiceCommand can be used as API for '<parameter>' with '<value>'" -TestCases @(
+    @{parameter="Force";value=$true},
+    @{parameter="Force";value=$false}
+  ) {
+    param($parameter, $value)
+    $restartservicecmd = [Microsoft.PowerShell.Commands.RestartServiceCommand]::new()
+    $restartservicecmd.$parameter = $value
+    $restartservicecmd.$parameter | Should Be $value
+  }
+
+  It "Stop/Start/Restart-Service works" {
+    $wasStopped = $false
+    try {
+      $spooler = Get-Service Spooler
+      $spooler | Should Not BeNullOrEmpty
+      if ($spooler.Status -ne "Running") {
+        $wasStopped = $true
+        $spooler = Start-Service Spooler -PassThru
+      }
+      $spooler.Status | Should Be "Running"
+      $spooler = Stop-Service Spooler -PassThru
+      $spooler.Status | Should Be "Stopped"
+      (Get-Service Spooler).Status | Should Be "Stopped"
+      $spooler = Start-Service Spooler -PassThru
+      $spooler.Status | Should Be "Running"
+      (Get-Service Spooler).Status | Should Be "Running"
+      Stop-Service Spooler
+      (Get-Service Spooler).Status | Should Be "Stopped"
+      $spooler = Restart-Service Spooler -PassThru
+      $spooler.Status | Should Be "Running"
+      (Get-Service Spooler).Status | Should Be "Running"
+    } finally {
+      if ($wasStopped) {
+        Stop-Service Spooler
+      }
+    }
+  }
+
+  It "Suspend/Resume-Service works" {
+    $workstation = Get-Service LanmanWorkstation
+    $workstation | Should Not BeNullOrEmpty
+    Suspend-Service LanmanWorkstation
```

Comment:
Can we use something else other than the workstation service?

Reply:
On my system, only 4 services support suspend/resume.  I'll use WER.

```powershell
PS C:\> get-service | ? { $_.CanPauseandcontinue -eq $true }

Status   Name               DisplayName
------   ----               -----------
Running  LanmanWorkstation  Workstation
Running  stisvc             Windows Image Acquisition (WIA)
Running  WerSvc             Windows Error Reporting Service
Running  Winmgmt            Windows Management Instrumentation
```